src/components/booklist/booklist.component.jsx

```diff
@@ -1,18 +1,28 @@
 import React from 'react';
 import Book from '../book/book.component';
 import { connect } from 'react-redux';
+import './booklist.styles.scss'
 
-const BookList = ({books}) => (
-	<div>
-		{ books.map(({id, title, category}) => (
-			<Book key={id} id={id} title={title} category={category} />
-		))
-		}
-	</div>
-)
+const BookList = ({books, filter}) => {
+	let filteredBooks
+	if (filter.length) {
+		filteredBooks = books.filter( ({category}) => category === filter )
+	} else {
+		filteredBooks = books
+	}
 
-const mapStateToProps = ({ library: {books} }) => ({
-	books
+	return (
+		<div className='booklist'>
+			{ filteredBooks.map(({id, title, category, read}) => (
+				<Book key={id} id={id} title={title} category={category} read={read} />
+			))
+			}
+		</div>
+)}
+
+const mapStateToProps = ({ library: {books}, filter: {filter} }) => ({
```

Comment:
Why is our `filter` argument like this: `filter: {filter})? Can't we simply:

```javascript
const mapStateToProps = (books, filter) => ({
 //
})
``` ?

Reply:
Well, not really, it throws an error if I refactor it to this format

Reply:
I mean, at least I could not get it working